openfe/setup/ligand_network_planning.py

```diff
@@ -210,3 +211,95 @@ def generate_minimal_spanning_network(
                            + str(list(missing_nodes)))
 
     return min_network
+
+
+def generate_network_from_names(
+        ligands: list[SmallMoleculeComponent],
+        mapper: AtomMapper,
+        names: list[tuple[str, str]],
+) -> LigandNetwork:
+    """Generate a LigandNetwork
+
+    Parameters
+    ----------
+    ligands : list of SmallMoleculeComponent
+      the small molecules to place into the network
+    mapper: AtomMapper
+      the atom mapper to use to construct edges
+    names : list of tuples of names
+      the edges to form where the values refer to names of the small molecules,
+      eg `[('benzene', 'toluene'), ...]` will create an edge between the
+      molecule with names 'benzene' and 'toluene'
+
+    Returns
+    -------
+    LigandNetwork
+
+    Raises
+    ------
+    KeyError
+      if an invalid name is requested
+    ValueError
+      if multiple molecules have the same name (this would otherwise be
+      problematic)
+    """
+    nm2idx = {l.name: i for i, l in enumerate(ligands)}
+
+    if len(nm2idx) < len(ligands):
+        dupes = Counter((l.name for l in ligands))
+        dupe_names = [k for k, v in dupes.items() if v > 1]
+        raise ValueError(f"Duplicate names: {dupe_names}")
+
+    try:
+        ids = [(nm2idx[nm1], nm2idx[nm2]) for nm1, nm2 in names]
+    except KeyError:
+        badnames = [nm for nm in itertools.chain.from_iterable(names)
+                    if nm not in nm2idx]
+        available = [ligand.name for ligand in ligands]
+        raise KeyError(f"Invalid name(s) requested {badnames}.  "
+                       f"Available: {available}")
```

Comment:
Not sure whether the long list of available names may swamp the short list of bad ones in practice. However, this might be a good place for fuzzy string matching, if we ever want to add that for usability help: "Unknown ligand 'benzalderp'. Did you mean 'benzaldehyde'?"

Not a blocker, but might be a useful todo to note in code.